src/enterprise/directory_sync/mod.rs

```diff
@@ -18,20 +18,37 @@ use super::{
 pub enum DirectorySyncError {
     #[error("Database error: {0}")]
     DbError(#[from] SqlxError),
-    #[error("Access token has expired")]
+    #[error("Access token has expired or is not present. An issue may have occured while trying to obtain a new one.")]
     AccessTokenExpired,
-    #[error("Failed to process request: {0}")]
-    RequestError(#[from] reqwest::Error),
-    #[error("Failed to build a JWT token: {0}")]
+    #[error("Processing a request to the provider's API failed: {0}")]
+    RequestError(String),
+    #[error(
+        "Failed to build a JWT token, required for communicating with the provider's API: {0}"
+    )]
     JWTError(#[from] jsonwebtoken::errors::Error),
     #[error("The selected provider {0} is not supported for directory sync")]
     UnsupportedProvider(String),
-    #[error("Directory sync not configured")]
+    #[error("Directory sync is not configured")]
     NotConfigured,
-    #[error("Defguard group not found: {0}")]
+    #[error("Couldn't map provider's group to a Defguard group as it doesn't exist. There may be an issue with automatic group creation. Error details: {0}")]
     DefGuardGroupNotFound(String),
 }
 
+impl From<reqwest::Error> for DirectorySyncError {
+    fn from(err: reqwest::Error) -> Self {
+        if err.is_decode() {
+            DirectorySyncError::RequestError(format!("There was an error while trying to decode provider's response, it may be malformed: {}", err))
```

Comment:
```suggestion
            DirectorySyncError::RequestError(format!("There was an error while trying to decode provider's response, it may be malformed: {err}))
```